theme_report/generate_theme_report.py

```diff
@@ -82,36 +74,45 @@ def main():
     # used for login page
     CANVAS_USER = ENV["CANVAS_USER"]
     CANVAS_PASSWORD = ENV["CANVAS_PASSWORD"]
-
-    # Initialize the driver
-    driver = webdriver.Chrome()
-
-    # Open provided link in a browser window using the driver
-    driver.get(API_URL + "/login/saml")
-    driver.execute_script(
-        f"document.getElementById('login').value = '{CANVAS_USER}';")
-    driver.execute_script(
-        f"document.getElementById('password').value = '{CANVAS_PASSWORD}';")
-    loginSubmit = driver.find_element_by_id('loginSubmit')
-    loginSubmit.click()
-
-    # create an empty dataframe
-    report_df = pd.DataFrame(
-        columns=('account_id', "account_name", "current_theme_name"))
+    browser = playwright.chromium.launch(headless=False)
+    context = browser.new_context()
+    page = context.new_page()
+    page.goto("https://canvas.it.umich.edu/")
```

Comment:
This should use the value from the env rather than have it hardcoded.

Actually I figured out how to bypass this entire weblogin process entirely for Canvas!

There's an API endpoint that can be used with API keys to get a time limited login.
https://canvas.instructure.com/doc/api/file.oauth_endpoints.html#get-login-session-token

I've modified this code to execute this to only use the API key! It's a hack for now but we can get it back into the CanvasAPI. I think it's really neat. If you're available tomorrow maybe we can talk about it.